### ISSUE TITLE
Dio does not remove content-length header when removing content-type header

### ISSUE THREAD

**Issue by ThinkDigitalSoftware** [flutterchina/dio#891]
### New Issue Checklist

- [X] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |      |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |      |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |   3.0.10   |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   | 100%     |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

Sorry, didn't have time to go through the entire process. But I'll still report the error and raise the PR

**Comment by ThinkDigitalSoftware** [comment]
Here's a PR I made to my own repo. I'm on a deadline and don't have the time to do all the steps required for the PR, so if anyone wants to implement the changes listed here and satisfy the pre-requisites, this can be fixed easily.
https://github.com/ThinkDigitalSoftware/dio/pull/1/commits/09de69cf2b7fe6b126987a8c9af5431204d167a6

**Comment by ThinkDigitalSoftware** [comment]
This isn't stale, just unfixed
